tests/testthat/test-classSelection.R

```diff
@@ -0,0 +1,133 @@
+  Y.cutoff <- factor(ifelse(Y.prob > .5, Y.levels[pos.level], Y.levels[-pos.level]), levels=Y.levels)
+
+  # Create confusion matricies for each class prediction vector
+  cmat.pred <- confusionMatrix(Y.pred, Y.test, positive=Y.levels[pos.level])
+  cmat.cutoff <- confusionMatrix(Y.cutoff, Y.test, positive=Y.levels[pos.level])
+
+  # Verify that the positive level of the Y response is equal to the positive
+  # class label used by caret.  This could potentially become untrue if
+  # the levels of the response were ever rearranged by caretEnsemble at some point.
+  expect_identical(cmat.pred$positive, Y.levels[pos.level])
+
+  # Verify that the accuracy score on predicted classes is relatively high.  This
+  # check exists to avoid previous errors where classifer ensemble predictions were
+  # being made using the incorrect level of the response, causing the opposite
+  # class labels to be predicted with new data.
+  expect_equal(as.numeric(cmat.pred$overall["Accuracy"]), 0.7586, tol = 0.0001)
+
+  # Similar to the above, ensure that probability predictions are working correctly
+  # by checking to see that accuracy is also high for class predictions created
+  # from probabilities
+  expect_equal(as.numeric(cmat.cutoff$overall["Accuracy"]), 0.7586, tol = 0.0001)
+}
+
+test_that("Ensembled classifiers do not rearrange outcome factor levels", {
+  skip_on_cran()
+
+  # Make sure that caretEnsemble uses the first level in the
+  # outcome factor as the target class
+  options(caret.ensemble.target.bin.level=1)
+
+  # First run the level selection test using the default levels
+  # of the response (i.e. c('No', 'Yes'))
+  set.seed(2239)
+  runBinaryLevelValidation(Y.train, Y.test)
+
+  # Now reverse the assigment of the response labels as well as
+  # the levels of the response factor.  Reversing the assignment
+  # is necessary to make sure the expected accuracy numbers are
+  # the same (i.e. Making a "No" into a "Yes" in the response means
+  # predictions of the first class will still be as accurate).
+  # Reversing the level order then ensures that the outcome is not
+  # releveled at some point by caretEnsemble.
+  Y.levels <- levels(Y.train)
+  refactor <- function(d) factor(
+    ifelse(d == Y.levels[1], Y.levels[2], Y.levels[1]),
+    levels=rev(Y.levels))
+
+  set.seed(2239)
+  runBinaryLevelValidation(refactor(Y.train), refactor(Y.test))
+
+})
+
+test_that("Target class selection configuration works", {
+  skip_on_cran()
+  set.seed(2239)
+  expect_equal(1, 1)
+
+  # Make sure that caretEnsemble uses the first level in the
+  # outcome factor as the target class
+  options(caret.ensemble.target.bin.level=2)
+
+  Y.levels <- levels(Y.train)
+  refactor <- function(d) factor(as.character(d), levels=rev(Y.levels))
+
+  set.seed(2239)
+  runBinaryLevelValidation(refactor(Y.train), refactor(Y.test), pos.level=2)
+
+  # Set the target class back to the default level of 1
+  options(caret.ensemble.target.bin.level=1)
```

Comment:
You should get the original option, and then re-set it at the end of the test, e.g.

```{R}
oldOption <- getOption('caret.ensemble.target.bin.level=')
...
options(caret.ensemble.target.bin.level=oldOption)
```

This prevents the test from changing the option, if the user has set it to something other than the default